src/storage-types/src/connections.rs

```diff
@@ -358,6 +368,166 @@ impl<C: ConnectionAccess> AlterCompatible for Connection<C> {
     }
 }
 
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum IcebergCatalogType {
+    Rest,
+    S3TablesRest,
+}
+
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct IcebergCatalogConnection<C: ConnectionAccess = InlinedConnection> {
+    /// What type of Iceberg catalog is being used
+    pub catalog_type: IcebergCatalogType,
+    /// Where the catalog is located
+    pub uri: String,
```

Comment:
During planning this is validated to be a `reqwest::Url`. Should we store it like this here?